server/index.js

```diff
@@ -0,0 +1,207 @@
+require('dotenv').config();
+const request = require('request');
+const http = require('http');
+const httpProxy = require('http-proxy');
+const express = require('express');
+const bodyParser = require('body-parser');
+const bcrypt = require('bcrypt');
+const querystring = require('querystring');
+const JWT = require('./jwt');
+
+const {
+  AIRTABLE_API_KEY,
+  AIRTABLE_BASE_ID,
+  AIRTABLE_ENDPOINT_URL,
+  AIRTABLE_API_VERSION,
+  AIRTABLE_USER_AGENT,
+  AIRTABLE_USER_TABLE,
+  HASH_PASSWORD,
+  SALT_ROUNDS
+} = process.env;
+
+const HEADERS = {
+  authorization: 'Bearer ' + AIRTABLE_API_KEY,
+  'x-api-version': AIRTABLE_API_VERSION,
+  'x-airtable-application-id': AIRTABLE_BASE_ID,
+  'User-Agent': AIRTABLE_USER_AGENT
+};
+
+const USERNAME_CREATE_URL = `${AIRTABLE_ENDPOINT_URL}/v0/${AIRTABLE_BASE_ID}/${AIRTABLE_USER_TABLE}?`;
+
+const app = express();
+
+const proxy = httpProxy.createProxyServer({
+  changeOrigin: true,
+  ignorePath: false
+});
+
+proxy.on('proxyReq', function(proxyReq, req, res, options) {
+  proxyReq.setHeader('authorization', `Bearer ${AIRTABLE_API_KEY}`);
+});
+
+app.use(bodyParser.json());
+app.use('/v0*', JWT.verifyToken);
```

Comment:
Should lines 43-49 be moved below the `/register` and `/login` routes (we don't want to JWT.verifyToken these ones)?

Reply:
The original `app.use('/v0*', JWT.verifyToken)` will only apply the middleware on the specified mounted path `/v0*`, even though it is applied before `/register` and `/login`.

`app.all('/v0*', JWT.verifyToken, (req, res) =>`

In the latest commit I changed it from `app.use` to `app.all` to make it clear we are only applying the `JWT.verifyToken` middleware only in `/v0*` paths at the application level (since we are applying the middleware using app.all vs the router itself)

Reply:
Whoops, forgot to mention that this is because `/login` and `/register` may have to go up to the `/v0` namespace (it should now exist at `/v0/:baseId/[login | register]`). This has to do with how URLs get built up in the client library.

Reply:
Moving the middleware down makes sense given the namespace point.

Before I make more modifications, we should probably decide if the client will be giving a `/login` and `/register` route or it will be `/v0/:baseId/[login | register]` or MAYBE something else?

I know that if the client implements a separate runAction, say `runActionAuth` then we'll be able to easily implement different paths `/login` vs `/v0/:baseId/login` vs `/airlock/v0/register`

**IDEA**: 
Do we want to have a seperate `airlock` namespace api route for airlock features/services? 
`/airlock/v0/[login | register]`

This is of course under the assumption that we would create our own runActionAuth/runActionAirlock function that can take care of our own custom pathing/namespace, just for `/login` and `register`

Approaching this with the mindset of separation of concerns, making it easier (hopefully) to integrate any new Airtable api change for the future (perhaps they rewrite their pathing/namespace/runAction), by making airlock pathing/namespace independent from airtable pathing/namespace this will be better for the future.

So I guess my question will be do we want `/login`, `register` vs `/v0/:baseId/[login | register] vs `/airlock/v0/[login | register]`

Reply:
I think we figured these out with the `__DANGEROUSLY_...` constants!